src/main/java/frc/robot/subsystems/drive/states/LinearDriveState.java

```diff
@@ -0,0 +1,266 @@
+package frc.robot.subsystems.drive.states;
+
+import coppercore.controls.state_machine.state.PeriodicStateInterface;
+import coppercore.controls.state_machine.transition.Transition;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import frc.robot.constants.JsonConstants;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Drive.DriveTrigger;
+import org.littletonrobotics.junction.Logger;
+
+public class LinearDriveState implements PeriodicStateInterface {
+  private Drive drive;
+
+  private Pose2d goalPose = null;
+
+  private double kDriveToPointTranslationP = 10;
+  private double kDriveToPointTranslationI = 0;
+  private double kDriveToPointTranslationD = 0;
+  private double kDriveTranslationMaxVelocity = 5;
+  private double kDriveTranslationMaxAcceleration = 5;
+  private double kPositionTolerance = 0.005;
+  private double kVelocityTolerance = 0.005;
+
+  private double kDriveToPointHeadingP = 10;
+  private double kDriveToPointHeadingI = 0;
+  private double kDriveToPointHeadingD = 0;
+  private double kDriveHeadingMaxVelocity = 5;
+  private double kDriveHeadingMaxAcceleration = 5;
+  private double kAngleTolerance = 0.005;
+  private double kAngularVelocityTolerance = 0.005;
```

Comment:
These should be made JSON configurable (probably new constants in DriveConstants.java/json)